drivers/flash/spi_nor.c

```diff
@@ -540,6 +546,104 @@ static int spi_nor_wrsr(const struct device *dev,
 	return ret;
 }
 
+#if DT_HAS_COMPAT_STATUS_OKAY(mxicy_mx25r_spi)
+
+/**
+ * @brief Read the configuration register.
+ *
+ * @note The device must be externally acquired before invoking this
+ * function.
+ *
+ * @param dev Device struct
+ *
+ * @return the non-negative value of the status register, or an error code.
+ */
+#define MX25R_CMD_RDCR 0x15
+static int mx25r_rdcr(const struct device *dev)
+{
+	uint16_t reg;
+	int ret = spi_nor_cmd_read(dev, MX25R_CMD_RDCR, &reg, sizeof(reg));
+
+	if (ret == 0) {
+		ret = reg;
+	}
+
+	return ret;
+}
+
+/**
+ * @brief Write the configuration register.
+ *
+ * @note The device must be externally acquired before invoking this
+ * function.
+ *
+ * @param dev Device struct
+ * @param cr  The new value of the configuration register
+ *
+ * @return 0 on success or a negative error code.
+ */
+static int mx25r_wrcr(const struct device *dev,
+			uint16_t cr)
+{
+	/* The configuration register bytes on the Macronix MX25R devices are
+	 * written using the Write Status Register command where the configuration
+	 * register bytes are written as two extra bytes after the status register.
+	 * First read out the current status register to preserve the value.
+	 */
+	int sr = spi_nor_rdsr(dev);
+
+	if (sr < 0) {
```

Comment:
`LOG_ERR` or  something, because it is hard to tell whether function failed here or in line 599, or 608, as all will just result in negative error value, and we completely rely on whatever called function returns with no mapping to the function specific codes.